Dockerfile.cloudwatcher

```diff
@@ -0,0 +1,56 @@
+###############################
+# STAGE 1 test and build binary
+###############################
+
+FROM golang:1.12 AS builder
+
+# Set Go build env vars
+ARG GOOS=linux
+ARG GOARCH=amd64
+ENV GOOS=${GOOS} GOARCH=${GOARCH}
+
+# Create a non-root user
+ARG GOUSER=appuser
+RUN adduser --gecos '' --disabled-password --no-create-home ${GOUSER}
+
+WORKDIR /go/src/app
+
+# Fetch dependencies
+# Do this first for caching sake
+ENV GO111MODULE=on
+COPY go.mod go.sum ./
+RUN go mod download
+
+# Copy source
+COPY . .
+
+# Install/Update packages (after src COPY so it always happens)
+RUN apt-get update -y \
+    && apt-get upgrade -y \
+    && apt-get install -y ca-certificates tzdata \
+    && update-ca-certificates
+
+# Build the binary
+RUN CGO_ENABLED=0 go build -ldflags='-w -s' -o /go/bin/cloudwatcher ./cmd/cloudwatcher
```

Comment:
What's the ldflags arg all about?  I see that 
`	-ldflags '[pattern=]arg list'
		arguments to pass on each go tool link invocation.
`
But where are they getting passed, exactly, and what do they signify?

Reply:
These make for a smaller Docker image. `-w` builds the binary without debugger information, and `-s` builds without the Go symbol table.
https://stackoverflow.com/questions/22267189/what-does-the-w-flag-mean-when-passed-in-via-the-ldflags-option-to-the-go-comman